src/fenics_constitutive/models/elasticity_laws.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from typing import Protocol
+
+import numpy as np
+
+from fenics_constitutive import StressStrainConstraint
+
+from .utils import lame_parameters
+
+
+class ElasticityLaw(Protocol):
+    def get_D(self, E: float, nu: float) -> np.ndarray: ...
+    def get_I2(self, stress_strain_dim: int) -> np.ndarray: ...
```

Comment:
a documentation would be helpful (D constitutive tangent, I2 identity matrix of rank 2 tensor?)